tex/chProgramming.tex

```diff
@@ -886,9 +890,10 @@ \subsection{Natural numbers}\index[concept]{natural number}
 \coqrun{name=larger_than4_run}{ssr,larger_than4}
 
 On the other hand, if we want to describe a different computation for
-three different cases and use variables in more than one case, we need
-the more general ``\C{match .. with .. end}'' syntax.  Here is an
-example:
+three different cases and use variables in more than one case, we
+must either awkwardly nest several ``\C{if .. then .. else}'' blocks
```

Comment:
I'm not in love with this one

Reply:
As far as I am concerned, only the "awkwardly" and "(better)" adverbs should be removed.

Reply:
I added them in to clarify that the "match" syntax is not strictly necessary here. Now that you're pointing this out, this is just as clear without the "awkwardly" and the "(better)". Good point.

Reply:
My problem is that if you nest ifs you don't obtain the same term. Maybe it's too technical, and the intuition you give is more important than this detail.

Reply:
Ah! Surprising. I'd say revert the change then, but I'd rather not create the impression that if-then-else is too weak for the problem at hand. Maybe rewrite as follows:
`The ``\C{if .. then .. else}'' syntax is just a particular case of the more general ``\C{match .. with .. end}'' construct, which allows us to separate several (not just two) cases.`

Reply:
I like this

Reply:
I like this too.